universal-application-tool-0.0.1/app/auth/UatProfileAdapter.java

```diff
@@ -55,7 +56,36 @@ public UatProfileAdapter(
       return Optional.empty();
     }
 
+    Optional<UatProfile> existingProfile = profileUtils.currentUserProfile(context);
```

Comment:
(nit) can you move this one line down. That way the `(OidcProfile) oidcProfile.get()` line is next to where it checks for `isEmpty` and `instanceof`.